src/main/scala/com/cloudera/sparkts/TimeSeriesUtils.scala

```diff
@@ -198,6 +195,36 @@ private[sparkts] object TimeSeriesUtils {
     }
   }
 
+  def rebaserGeneric(
+      sourceIndex: DateTimeIndex,
+      targetIndex: DateTimeIndex,
+      defaultValue: Double): Vector[Double] => Vector[Double] = {
+    // indexMapping(i) = j means that resultArr(i) should be filled with the value from
+    // vec(j)
+    val indexMapping = new Array[Int](targetIndex.size)
+
+    val targetIndexZDTIterator = targetIndex.zonedDateTimeIterator
+    var i = 0
+    while (targetIndexZDTIterator.hasNext) {
+      indexMapping(i) = sourceIndex.locAtDateTime(targetIndexZDTIterator.next)
+      i += 1
+    }
+
+    vec: Vector[Double] => {
```

Comment:
Given that this shares an implementation with the one used in `rebaserIrregularSourceIrregularTarget`, can we break it out into a separate helper function?